src/main/scala/com/amazon/deequ/analyzers/applicability/Applicability.scala

```diff
@@ -133,34 +146,38 @@ private[deequ] class Applicability(session: SparkSession) {
     * @param check A check that may be applicable to some data
     * @param schema The schema of the data the checks are for
     */
-  def isApplicable(check: Check, schema: StructType): ApplicabilityResult = {
+  def isApplicable(check: Check, schema: StructType): CheckApplicability = {
 
     val data = generateRandomData(schema, 1000)
 
-    val constraintsByName = check.constraints
+    val namedMetrics = check.constraints
       .map { constraint => constraint.toString -> constraint }
       .map {
         case (name, nc: ConstraintDecorator) => name -> nc.inner
         case (name, c: Constraint) => name -> c
       }
       .collect { case (name, constraint: AnalysisBasedConstraint[_, _, _]) =>
-        (name, constraint)
+        (name, constraint.analyzer.calculate(data).value)
       }
 
+    val constraintApplicabilities = check.constraints.zip(namedMetrics).map {
+      case (constraint, (_, metric)) =>
+        constraint -> metric.isSuccess
+    }.toMap
```

Comment:
.toMap should be on the next line

Reply:
Still haven't lost hope for myself that I'll learn deequ style :)

Do you think we could have a rule for it in our style check?

Reply:
We tried to configure such rules, unfortunately, there is always an edge case that is not coverable by the stylechecker...